### ISSUE TITLE
Unable to build the project

### ISSUE THREAD

**Issue by jonjomckay** [manywho/ui-core#3]
When running the `gulp dev-ts --build=../flow-test-project/www` command, the build fails due to some kind of type mismatch:

`/Users/jonjo/ManyWho/ui-cordova/ui-core/typings/globals/react/index.d.ts(159,11): error TS2559: Type 'Component<P, S>' has no properties in common with type 'ComponentLifecycle<P, S>'.`

I also get the same error when trying to run the same command in the `ui-offline` project, so I think it's something to do with the dependency on `typings` (which has been throwing a bunch of deprecation errors, so we shouldn't be using it).

**Comment by sudarshana** [comment]
@matt-manywho @jonjomckay Not quite sure what needs to happen to get this working.. may we treat this as a priority please? Be great to have the documentation of this shipped while Jonjo is here - if possible. :) Thank you.

**Comment by matt-manywho** [comment]
@jonjomckay That shouldn't stop the build from completing as its a tslint error, I've tried it here and the "compiled" javascript is still emitted. It is annoying, its happening essentially due to our React dependency being out of date. This, `ui-core` and `ui-bootstrap` are currently in the process of being updated to webpack, React 16, etc so this will go away soon

**Comment by jonjomckay** [comment]
@matt-manywho you're right, the compiled scripts are put into the correct folder - the command output is pretty confusing though. Running it today in the `ui-offline` project at least, with 0 changes, gives me this output (the missing whitespace messages actually number so many that it reaches the maximum height for my terminal history):

```
(whitespace) /Users/jonjo/ManyWho/ui-cordova/ui-offline/js/services/metadata.ts[1, 25806]: missing whitespace
(whitespace) /Users/jonjo/ManyWho/ui-cordova/ui-offline/js/services/metadata.ts[1, 25817]: missing whitespace
(whitespace) /Users/jonjo/ManyWho/ui-cordova/ui-offline/js/services/metadata.ts[1, 25825]: missing whitespace
(whitespace) /Users/jonjo/ManyWho/ui-cordova/ui-offline/js/services/metadata.ts[1, 25836]: missing whitespace
(whitespace) /Users/jonjo/ManyWho/ui-cordova/ui-offline/js/services/metadata.ts[1, 25841]: missing whitespace
(whitespace) /Users/jonjo/ManyWho/ui-cordova/ui-offline/js/services/metadata.ts[1, 25852]: missing whitespace

[16:47:12] [gulp-tslint] Failed to lint: 3272 errors.
/Users/jonjo/ManyWho/ui-cordova/ui-offline/typings/globals/react/index.d.ts(159,11): error TS2559: Type 'Component<P, S>' has no properties in common with type 'ComponentLifecycle<P, S>'.
[16:47:14] TypeScript: 1 semantic error
[16:47:14] TypeScript: emit succeeded (with errors)
[16:47:14] Finished 'dev-ts' after 3.01 s
jonjo@kaylie  ~/ManyWho/ui-cordova/ui-offline  develop ?                                                                                    1 ↵  10017  16:48:34
```

I think those should be fixed as otherwise it's not going to be a very clear experience for users as it looks like a lot of stuff is broken...
